internal/controller/bucket_controller_test.go

```diff
@@ -0,0 +1,446 @@
+package controller
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/giantswarm/object-storage-operator/api/v1alpha1"
+	"github.com/giantswarm/object-storage-operator/internal/pkg/managementcluster"
+	"github.com/giantswarm/object-storage-operator/internal/pkg/service/objectstorage/objectstoragefakes"
+)
+
+var _ = Describe("Bucket Reconciler", func() {
+	const (
+		BucketName      string = "my-bucket-name"
+		BucketNamespace string = "default"
+	)
+
+	var (
+		ctx context.Context
+
+		reconciler   BucketReconciler
+		reconcileErr error
+
+		fakeClient     client.Client
+		serviceFactory objectstoragefakes.FakeObjectStorageServiceFactory
+		service        objectstoragefakes.FakeObjectStorageService
+		bucketKey      = client.ObjectKey{
+			Name:      BucketName,
+			Namespace: BucketNamespace,
+		}
+	)
+
+	// creates the dummy bucket and clients
+	BeforeEach(func() {
+		SetDefaultEventuallyPollingInterval(time.Second)
+		SetDefaultEventuallyTimeout(time.Second * 90)
+
+		ctx = context.Background()
+
+		serviceFactory = objectstoragefakes.FakeObjectStorageServiceFactory{}
+		service = objectstoragefakes.FakeObjectStorageService{}
+		serviceFactory.NewS3ServiceReturns(&service, nil)
+	})
+
+	var _ = Describe("CAPA", func() {
+		// creates the reconciler
+		BeforeEach(func() {
+			fakeClient = fake.NewClientBuilder().WithStatusSubresource(&v1alpha1.Bucket{}).Build()
```

Comment:
Using the k8sClient from the suite, I get errors because it's not aware of the CAPA scheme. Is there a way I could use the client without adding the CAPA scheme (and dependency) as I'm using unstructed ?